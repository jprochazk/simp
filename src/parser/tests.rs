use std::fmt::Write as _;
use std::path::Path;

fn parse(code: &str) -> String {
    match super::parse(code) {
        Ok(ast) => {
            // we want slightly smaller snapshots, so debug-fmt the contents directly
            let mut o = String::new();
            for stmt in &ast.body {
                writeln!(&mut o, "{stmt:#?}").ok();
            }
            if let Some(tail) = &ast.tail {
                writeln!(&mut o, "{tail:#?}").ok();
            }
            o
        }
        Err(err) => format!("{}", err.render(code)),
    }
}

#[glob_test::glob("./inputs/**/*.simp")]
fn test_parse(path: &Path) {
    let input = std::fs::read_to_string(path).unwrap();
    insta::assert_snapshot!(parse(&input));
}
