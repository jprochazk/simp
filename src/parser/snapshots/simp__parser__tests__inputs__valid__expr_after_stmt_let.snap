---
source: src/parser/tests.rs
expression: parse(&input)
---
StmtLet {
    name: "v",
    value: ExprInt {
        value: 0,
    },
}
ExprInt {
    value: 0,
}
